cpp/src/arrow/testing/gtest_util_test.cc

```diff
@@ -292,6 +292,7 @@ TEST(AssertTestWithinUlp, Basics) {
   EXPECT_FATAL_FAILURE(AssertWithinUlp(123.456f, 123.456085f, 10), "not within 10 ulps");
 }
 
+#ifdef ARROW_COMPUTE
```

Comment:
while this might solve the specific test failure, I think the underlying problem is not solved.
`gtest_util.cc` is build if `ARROW_TESTING=ON`, it should not require run_end_encode as that requires `ARROW_COMPUTE`.
In my opinion the `RunEndEncodeTableColumns` function should be part of test utilities from `arrow::compute`, I created a PR for that:
https://github.com/apache/arrow/pull/46161
This also appears on other issues as the one found here:
https://github.com/apache/arrow/pull/45618/files#diff-8e690315a407919d147f16a9a534fc9f5c3f6345b58820e2057afa7cd14bea6bR741
were I had to add run_end_encode to the non-conditional sources that are always included as part of libarrow as a workaround.